components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/messaging_reference.adoc

```diff
@@ -0,0 +1,589 @@
+
+[discrete]
+==== Function definition
+
+Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol[`StarknetMessaging.sol`]
+
+* Function: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol#L161[`cancelL1ToL2Message`]
+
+'''
+
+[#consumeMessageFromL2]
+=== `consumeMessageFromL2`
+
+[discrete]
+==== Description
+
+Consumes a message that was sent from an L2 contract.
+
+Returns the hash of the message.
+
+[discrete]
+==== State Mutability
+
+None.
+
+[discrete]
+==== Parameters
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`uint256 _fromAddress_`:: The address of the L2 contract sending the message.
+`uint256[] calldata _payload_`:: The payload of the message.
+
+[discrete]
+==== Returns
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`bytes32 _msgHash_`:: The hash of the consumed message.
+
+[discrete]
+==== Function definition
+
+Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol[`StarknetMessaging.sol`]
+
+* Function: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol#L132[`consumeMessageFromL2`]
+
+'''
+
+[#getMaxL1MsgFee]
+=== `getMaxL1MsgFee`
+
+[discrete]
+==== Description
+
+Returns the maximum fee, in Wei, that Starknet #charges# #(Is "charges" correct? It says "accepts" link:https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/starknet/solidity/IStarknetMessaging.sol#L8[in the source file])# for a single message.
+
+[discrete]
+==== State Mutability
+
+`pure`
+
+[discrete]
+==== Parameters
+
+None.
+
+[discrete]
+==== Returns
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`uint256 _MAX_L1_MSG_FEE_`:: The maximum fee, in Wei, that Starknet #accepts/charges?# per single message.
+
+[discrete]
+==== Function definition
+
+Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol[`StarknetMessaging.sol`]
+
+* Function: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol#L31[`getMaxL1MsgFee`]
+
+'''
+
+[#l1ToL2MessageCancellations]
+=== `l1ToL2MessageCancellations`
+
+[discrete]
+==== Description
+
+Returns the timestamp at the time cancelL1ToL2Message was called with a given message hash.
+
+[discrete]
+==== State Mutability
+
+`view`
+
+[discrete]
+==== Parameters
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`bytes32 _msgHash_`:: The message hash.
+
+[discrete]
+==== Returns
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`uint256 _result_`:: The timestamp at the time cancelL1ToL2Message was called with the given message hash `_msgHash_`. Returns 0 if cancelL1ToL2Message was never called with the given message hash `_msgHash_`.
```

Comment:
```suggestion
`uint256 _result_`:: The timestamp, in seconds since the Linux epoch, at the time `cancelL1ToL2Message` was called with the message hash `_msgHash_`. 

Returns `0` if `cancelL1ToL2Message` was not called with the message hash `_msgHash_`.
```

Reply:
Good